sanity/schemas/planningApplication.ts

```diff
@@ -163,6 +163,22 @@ export default defineType({
           description: "please select the status of the stage",
           name: "status",
           type: "object",
+          validation: (Rule: any) =>
```

Comment:
![image](https://github.com/user-attachments/assets/522cfa4f-77f9-41fe-9a87-3092c7a5ce49)

I was getting the above validation error instead of the custom message. I think it's because we need to check that stage and status exist before we access them. We could do something like this: 


```
          validation: (Rule: any) =>
            Rule.custom((field: any, context: any) => {
              const currentStage =
                context.document?.applicationStage?.stage?.toLowerCase();
              if (!currentStage) {
                return true;
              }

              if (!field || field[currentStage] === undefined) {
                return "This field must not be empty.";
              }

              return true;
            }),
```